pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -676,14 +678,25 @@ protected static Expression invokeCompileTimeFunctionExpression(Expression funcE
       try {
         FunctionInvoker invoker = new FunctionInvoker(functionInfo);
         Object result = invoker.process(arguments);
-        if (result instanceof String) {
-          result = String.format("'%s'", result);
-        }
         return RequestUtils.getLiteralExpression(result);
       } catch (Exception e) {
         throw new SqlCompilationException(new IllegalArgumentException("Unsupported function - " + funcName, e));
       }
     }
     return funcExpr;
   }
+
+  public static boolean isLiteralOnlyExpression(Expression e) {
+    if (e.getType() == ExpressionType.LITERAL) {
+      return true;
+    }
+    if (e.getType() == ExpressionType.FUNCTION) {
+      Function functionCall = e.getFunctionCall();
+      if (functionCall.getOperator().equalsIgnoreCase(SqlKind.AS.toString())) {
```

Comment:
Does this work only with `AS`?

Reply:
Yes, the CalciteSQLParser already evaluated the transform function with literals and replace it with literal.
So the only left over is to handle here is `Literal` and `Function` `AS`.